test/integration/auth/svcaccttoken_test.go

```diff
@@ -291,6 +319,166 @@ func TestServiceAccountTokenCreate(t *testing.T) {
+				t.Fatalf("expected err creating token for nonexistant svcacct but got: %#v", resp)
+			}
+			warningHandler.assertEqual(t, nil)
+			sa, del := createDeleteSvcAcct(t, cs, sa)
+			defer del()
+
+			warningHandler.clear()
+			if resp, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{}); err == nil {
+				t.Fatalf("expected err creating token bound to nonexistant pod but got: %#v", resp)
+			}
+			warningHandler.assertEqual(t, nil)
+			pod, delPod := createDeletePod(t, cs, scheduledpod)
+			defer delPod()
+
+			if node != nil {
+				var delNode func()
+				node, delNode = createDeleteNode(t, cs, node)
+				defer delNode()
+			}
+
+			// right uid
+			treq.Spec.BoundObjectRef.UID = pod.UID
+			warningHandler.clear()
+			if _, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{}); err != nil {
+				t.Fatalf("err: %v", err)
+			}
+			warningHandler.assertEqual(t, nil)
+			// wrong uid
+			treq.Spec.BoundObjectRef.UID = wrongUID
+			warningHandler.clear()
+			if resp, err := cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{}); err == nil {
+				t.Fatalf("expected err creating token bound to pod with wrong uid but got: %#v", resp)
+			}
+			warningHandler.assertEqual(t, nil)
+			// no uid
+			treq.Spec.BoundObjectRef.UID = noUID
+			warningHandler.clear()
+			treq, err = cs.CoreV1().ServiceAccounts(sa.Namespace).CreateToken(ctx, sa.Name, treq, metav1.CreateOptions{})
+			if err != nil {
+				t.Fatalf("err: %v", err)
+			}
+			warningHandler.assertEqual(t, nil)
+
+			checkPayload(t, treq.Status.Token, `"system:serviceaccount:myns:test-svcacct"`, "sub")
+			checkPayload(t, treq.Status.Token, `["api"]`, "aud")
+			checkPayload(t, treq.Status.Token, `"test-pod"`, "kubernetes.io", "pod", "name")
+			checkPayload(t, treq.Status.Token, "null", "kubernetes.io", "secret")
+			checkPayload(t, treq.Status.Token, `"myns"`, "kubernetes.io", "namespace")
+			checkPayload(t, treq.Status.Token, `"test-svcacct"`, "kubernetes.io", "serviceaccount", "name")
+			if node != nil {
+				checkPayload(t, treq.Status.Token, fmt.Sprintf(`"%s"`, node.Name), "kubernetes.io", "node", "name")
+			} else {
+				checkPayload(t, treq.Status.Token, "null", "kubernetes.io", "node")
+			}
+
+			expectedExtraValues := map[string]authenticationv1.ExtraValue{
+				"authentication.kubernetes.io/pod-name": {pod.ObjectMeta.Name},
+				"authentication.kubernetes.io/pod-uid":  {string(pod.ObjectMeta.UID)},
+			}
+			// we only expect to have node name and uid included if the node exists
+			if node != nil {
+				expectedExtraValues["authentication.kubernetes.io/node-name"] = authenticationv1.ExtraValue{node.ObjectMeta.Name}
+				expectedExtraValues["authentication.kubernetes.io/node-uid"] = authenticationv1.ExtraValue{string(node.ObjectMeta.UID)}
+			}
```

Comment:
We need to be consistent with checking for the node UID.  Also by making it optional, I am wondering if this test will occasionally flake.

Reply:
Fixed :) PTAL